api/src/core/usecases/importFromSource.ts

```diff
@@ -24,19 +24,25 @@ export const importFromSource: (dbApi: DbApiV2) => ImportFromSource = (dbApi: Db
               });
 
         const getSoftwareForm = resolveAdapterFromSource(source).softwareForm.getById;
+        let result = [];
 
         switch (source.kind) {
             case "HAL":
                 // Get All or Request
                 const rawHALSoftwareIds =
-                    softwareIdOnSource && softwareIdOnSource.length > 0
+                    softwareIdOnSource && softwareIdOnSource.length > 0 && softwareIdOnSource[0] !== ""
                         ? softwareIdOnSource
                         : (await halAPIGateway.software.getAll({ SWHFilter: true })).map(doc => doc.docid);
 
                 console.info(
                     `[UC:Import] Importing  ${rawHALSoftwareIds.length} software packages from ${source.slug}`
                 );
-                return rawHALSoftwareIds.map(docId => checkSoftware(dbApi, source, docId, getSoftwareForm, agentId));
+
+                for (const docId of rawHALSoftwareIds) {
+                    const newId = await checkSoftware(dbApi, source, docId, getSoftwareForm, agentId);
+                    result.push(newId);
+                }
+                return Promise.resolve(result.filter(val => val != undefined));
```

Comment:
as function is async, you can do just :

```suggestion
                return result.filter(val => val != undefined);
```